src/idl_gen_grpc.cpp

```diff
@@ -165,53 +171,260 @@ class FlatBufFile : public grpc_generator::File {
+    : parser_(parser), file_name_(file_name) {}
+  FlatBufFile &operator=(const FlatBufFile &);
+
+  std::string filename() const {
+    return file_name_;
+  }
+  std::string filename_without_ext() const {
+    return StripExtension(file_name_);
+  }
+
+  std::string package() const {
+    return StripExtension(parser_.namespaces_.back()->GetFullyQualifiedName(""));
+  }
+
+  std::vector<std::string> package_parts() const {
+    return parser_.namespaces_.back()->components;
+  }
+
+  //un-used
+  std::string additional_headers() const {
+    return "import \"github.com/google/flatbuffers/go\"";
+  }
+
+  //un-used
+  std::string message_header_ext() const { return ""; }
+  std::string service_header_ext() const { return ""; }
+
+  int service_count() const {
+    return static_cast<int>(parser_.services_.vec.size());
+  };
+
+  std::unique_ptr<const grpc_generator::Service> service(int i) const {
+    return std::unique_ptr<const grpc_generator::Service> (
+             new FlatBufService(parser_.services_.vec[i]));
+  }
+
+  std::unique_ptr<grpc_generator::Printer> CreatePrinter(std::string *str) const {
+    return std::unique_ptr<grpc_generator::Printer>(
+             new FlatBufPrinter(str));
+  }
+
+private:
+  const Parser &parser_;
+  const std::string &file_name_;
+};
+
+std::string GetCodec() {
+  std::string codec;
+  auto printer = std::unique_ptr<FlatBufPrinter>(new FlatBufPrinter(&codec));
+  printer->Print("// FlatCodec implements Grpc-go Codec\n\n");
+  printer->Print("type FlatCodec struct{}\n\n");
+  printer->Print("func (FlatCodec) Marshal(v interface{}) ([]byte, error) {\n");
+  printer->Indent();
+  printer->Print("return v.(*flatbuffers.Builder).FinishedBytes(), nil\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("func (FlatCodec) Unmarshal(data []byte, v interface{}) error {\n");
+  printer->Indent();
+  printer->Print("v.(flatbuffersInit).Init(data, flatbuffers.GetUOffsetT(data))\n");
+  printer->Print("return nil\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("func (FlatCodec) String() string {\n");
+  printer->Indent();
+  printer->Print("return \"flatbuffers\"");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("type flatbuffersInit interface {\n");
+  printer->Indent();
+  printer->Print("Init(data []byte, i flatbuffers.UOffsetT)\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  return codec;
+}
+
+bool GenerateGoGRPC(const Parser &parser,
+                    const std::string &file_name) {
+
+  FlatBufFile goFile(parser, file_name);
+  grpc_go_generator::Parameters p;
+  p.custom_codec_source = GetCodec();
+  p.custom_method_input = "flatbuffers.Builder";
+  std::string output = grpc_go_generator::GenerateServiceSource(&goFile, &p);
+  return flatbuffers::SaveFile((goFile.package() +"/" "grpc.go").c_str(),output, false);
+}
+
+}//namespace grpc_go
+
+bool GenerateGRPC(const Parser &parser,
```

Comment:
No, I meant rename this function with Go, since it appears Go specific. Then you can inline the function above.